### DISCUSSION TITLE
Polyglot - Userscript for Codewars

### DISCUSSION THREAD

**Discussion by hobovsky** [codewars/codewars.com#2377] [Show and tell]
Hello all,

Some time ago I've created a Tampermonkey Userscript which would help me with some issues which were particularly annoying for me, especially addressing issues #790, #1162, #1272, #1641 and indirectly #727, #752. A bunch of users already uses it, so maybe others would also find it useful. Feel free to use it, or to completely ignore it ;)

You can find the script in [this repo](https://github.com/hobovsky/polyglot).

Small summary of features:

1. Solved languages are marked:
![image](https://user-images.githubusercontent.com/23709795/54123489-36b48600-4400-11e9-8d41-8deec3259474.png)

1. Solved languages are dimmed also in language dropdowns, so you do not select the same language twice!
![image](https://user-images.githubusercontent.com/23709795/54272587-4957c800-4584-11e9-998f-21f0188f909a.png)

1. Filtering by "Solved/not solved in [language]":
![image](https://user-images.githubusercontent.com/23709795/54123650-8bf09780-4400-11e9-9e66-9930d5f29bf0.png)

1. Copy code to clipboard
![image](https://user-images.githubusercontent.com/23709795/54124041-8e072600-4401-11e9-8efb-4b00352dfee8.png)

1. Solutions are tabbed by language
![image](https://user-images.githubusercontent.com/23709795/54124157-d0306780-4401-11e9-88d4-dd28b82080bc.png)

1. Spoiler flags are always displayed, so it's more difficult to accidentally flag/unflag a comment.

Hopefully, some/all of the features will be one day incorporated into Codewars website itself. I will also add new features which would generally aim at improving user experience and fix small annoyances. If anyone of you finds the script useful, then great. If not, well, you probably are not annoyed by these issues, so that's also great!

Have fun!

**Comment by hobovsky** [comment]
## New feature: Rank leaderboards

Since Codewars now provides API endpoints for rank leaderboards but no GUI, the Overall and language specific rank boards can now be viewed with the script installed:

![image](https://user-images.githubusercontent.com/23709795/167502281-955bd6e5-8e75-4128-847e-c37948fc0d1f.png)

The feature is to be improved yet, but it seems to work quite well!

**Reply by hobovsky** [reply]
This one has been implemented directly in Codewars, and removed from Polyglot.

**Comment by hobovsky** [comment]
## New feature: rank assessments breakdown

Since Codewars now provides REST API for fetching rank assessments but does not have any GUI for this, the script can be used to view breakdown charts, hopefully helping to pick the most appropriate rank when approving kata.

![image](https://user-images.githubusercontent.com/23709795/172865627-db13f2d4-0de3-4139-a80c-ccba23cf904c.png)

**Reply by galalmuhammed** [reply]
nice

**Comment by lakritsbanan** [comment]
Has codewars website changed a lot since this that some of the functionality of the script doesn't work anymore? When I list the katas I have solved I don't get any options to filter the results by solved in a specific language. The script seems to be working though since I get the little "copy to clipboard" in the code boxes. Is the problem on my end or does this filter functionality not exist anymore?

**Reply by hobovsky** [reply]
Yes, the website has changed and some functions and features which previously were provided by the script are now available directly on the platform, like markers for completed languages, rank leaderboards, and filtering by "solved in language". Now you can see all challenges you solved in JS by selecting "language: JavaScript" and "progress: kata I have completed". This will show you only challenges you completed in JS.

**Reply by lakritsbanan** [reply]
You're right. I can't believe I didn't realize that. Maybe I got confused by the listing of kata solutions on the profile page. That's still the easiest way to just browse through them, and I don't think it's possible to sort or filter those on language.

**Reply by lakritsbanan** [reply]
Thanks for the rapid feedback.

**Comment by hobovsky** [comment]
## New feature: notifications when you rank up on a leaderboard

<img width="1066" height="284" alt="image" src="https://github.com/user-attachments/assets/1e786fa3-e6d3-4235-8396-5092a9798809" />

Implemented by @Kacarott
